packages/scene-system/sdk/SceneRuntime.ts

```diff
@@ -533,7 +534,28 @@ export abstract class SceneRuntime extends Script {
       }
 
       try {
-        await this.runCode(source as any as string, { dcl })
+        const self = this
+        class RestrictedWebSocket extends WebSocket {
```

Comment:
I'd like this isn't here, but I need the access to Permissions API, or maybe a promise which can access to it

Reply:
Maybe we can move this RestrictedWebScoket to a new file and create a fn to construct the permisisons. Something like



## SceneRuntime.ts
```ts
  const { Permissions } = await this.loadAPIs(['Permissions'])
  const restrictedWebSocket = createWebSocket(Permissions)
```
## WebSocket.ts
```ts
function createWebSocket(permission: Permissions) {
  return class RestrictedWebSocket extends WebSocket {
    constructor(url: string | URL, protocols?: string | string[]) {
      console.log({ url })
      if (url.toString().toLowerCase().substr(0, 4) !== 'wss:') {
...etc
```

or if you dont want to block the sceneRuntime you can pass a fn to createWebSocket

```ts
        async function getPermissions() {
          const { Permissions } = self.loadAPIs(['Permissions'])
          return Permissions
        }
```
And at `websocket.ts` call this async fn and then check if it has the valid permissions.
```ts
getPermissions().then(permission => permission.hasPermission....)
```